### ISSUE TITLE
Text size of label resets to default 16 when open label edit dialog

### ISSUE THREAD

**Issue by Mingun** [shutter-project/shutter#301]
#### Brief summary of issue
If you add text label on screenshot, change font size, close and reopen label dialog, then size resets back to its default value 16. See the video attached

#### Steps to reproduce the issue

1. Add text label
2. Select font size
3. Click OK
4. Double click on text label
5. Notice, that font size reset to the default value

#### Extra information, such as Shutter version, operating system and ideas for how to solve:

**Version**: 0.94.3 (from Ubuntu 20.04 repositories)
**Revision**: 1306

See the video: https://user-images.githubusercontent.com/450131/105798395-df980a00-5fb3-11eb-8880-c4689002db9b.mp4

**Comment by DarthGandalf** [comment]
Confirmed

**Comment by Mingun** [comment]
With release 0.97 (Rev.1549) the situation made even worse: now the using text tool starts with no selected font and zero size:
![No font and zero size](https://user-images.githubusercontent.com/450131/126443675-225b5748-916b-40c6-a325-7e5137e4ff13.png)

The font on the bottom toolbar displayed correctly:

https://user-images.githubusercontent.com/450131/126627776-bde4ff4b-bb3d-4049-9fee-d118bb74ab04.mp4

_(re-uploaded the video. According to the https://github.com/ccrisan/motioneye/issues/1067 Firefox does not support YUV444, which [kazam](https://github.com/hzbd/kazam) used to encode the video)_

**Comment by DarthGandalf** [comment]
> With release 0.97 (Rev.1549) the situation made even worse

Yep :( I had to mitigate the crash introduced in 0.96

Though "0" there is unexpected, should have been just the default value instead.

P.S. your video doesn't load, says file is damaged?

**Comment by Mingun** [comment]
Is there any obvious fix to at least return as it was before? Current behavior is very annoying.

**Comment by DarthGandalf** [comment]
Use the previous version for now?

**Comment by Mingun** [comment]
Unfortunately Ubuntu 20.04 contains only one version of the package so it will require dancing with a tambourine again: find old package, its dependencies... Maybe I have to use previous version but I'll try to look at the code first. What commit, as you think, could have broken that?

**Comment by DarthGandalf** [comment]
51cf4e2492de8e87c4ff67a1e0988a96df7e9d90 is most likely

**Comment by wkevin** [comment]
```
sudo vi /usr/share/perl5/Shutter/Draw/DrawingTool.pm
```

To line 4198, modify 

```
my ($font_desc) = $attr_list->get_iterator->get_font;
```

to

```
my ($font_desc) = Pango::FontDescription->from_string($self->{_font});
```

It works in my ubuntu20.04。

**Comment by TomlDev** [comment]
@wkevin great! 🚀  Works also on Linux Mint 20 with Shutter 0.99.2 ❤️

**Comment by Photon89** [comment]
I think, the problem is, that the TextTagTable which should contain the font information for all text fields is newly created each time that a text field is being edited: 

https://github.com/shutter-project/shutter/blob/132258e267461ebbc2fceb5b350f530c33d5fe42/share/shutter/resources/modules/Shutter/Draw/DrawingTool.pm#L4717

Thus, font information is only used once for creating the text but then not stored anywhere. So it takes the font information from the global font setting (which can be set in the bottom tool properties panel):

https://github.com/shutter-project/shutter/blob/132258e267461ebbc2fceb5b350f530c33d5fe42/share/shutter/resources/modules/Shutter/Draw/DrawingTool.pm#L4706

The TextTagTable should be a global object and contain font information from all text fields in the current Drawing Tool session. The global font setting button should show the values for the currently selected text field, which it has obtained from the TextTagTable.

It seems like quite some changes are required to implement this, though...

**Comment by Mingun** [comment]
For now I use simple workaround: instead of click to place label, click and drag. That will create a text with default label and correct font

**Comment by Photon89** [comment]
Could you elaborate on this? Which is the correct font in this case, if you are creating a new label? I thought, the issue is that the font isn't correctly showing for an existing label which is being edited.

**Comment by Mingun** [comment]
I do not know, which font should be considered as correct. I suspect, that the intention is to use last used font size, or use the default font size. I do not know, I always use default size. The click-and-drag technique fills font parameters like they was filled in 0.94.3 with just click.

**Comment by Photon89** [comment]
Are you possibly still using version 0.97? As in the latest stable version the behavior should be exactly like it was in 0.94.3 (what you described in the initial bug report).

**Comment by Mingun** [comment]
No, I'm using 0.99.2 rev 1593:
[Shutter 0.99.2-rev1593.webm](https://github.com/shutter-project/shutter/assets/450131/64eca081-1005-4eb2-a8b9-e69b4ca726b6)

**Comment by Photon89** [comment]
Right, it isn't fixed there yet. It should be fixed in later versions though, in particular in the newly released 0.99.5, at least to the extent that you described in the original bug report.
